File: dotnet/cafdemalihapa/Applications/DirectoryManagement/DirectoryOpenStrategies/MacDirectoryOpenStrategy.cs
using System.Runtime.InteropServices;
using Microsoft.Extensions.Logging;

namespace cafdemalihapa.Applications.DirectoryManagement.DirectoryOpenStrategies
{
    public class MacDirectoryOpenStrategy(ILogger<MacDirectoryOpenStrategy> logger)
        : DirectoryOpenStrategy(logger)
    {
        protected override string FileName => "open";

        public override bool CanHandle() => RuntimeInformation.IsOSPlatform(OSPlatform.OSX);
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Applications/DirectoryManagement/DirectoryOpenStrategies/MacDirectoryOpenStrategy.fs
namespace cafdemwimu.console.Applications.DirectoryManagement.DirectoryOpenStrategies

open System.Runtime.InteropServices
open Microsoft.Extensions.Logging

type MacDirectoryOpenStrategy(logger: ILogger<MacDirectoryOpenStrategy>) =
    inherit DirectoryOpenStrategy(logger)

    override _.FileName = "open"

    override _.CanHandle() = RuntimeInformation.IsOSPlatform(OSPlatform.OSX)
